### DISCUSSION TITLE
filter query with variable as operator

### DISCUSSION THREAD

**Discussion by LuffyBoy69** [sqlalchemy/sqlalchemy#10423] [Usage Questions]
Hello,

I want to know if it's possible to pass a variable as operator in a filter query ?

field = 'foo'
operator = 'contains'
value = 'bar'

 query = query.filter(db.or_(getattr(eval(table), field).operator(value))) 

Regards,

**Comment by CaselIT** [comment]
Hi,

This question is a bit vague. Is the operator an arbitrary string or is the name of a method of a column?

If it's a name of a column method, you could do something like:
```py
model = globals()[model_name]
col = getattr(model, field)
clause = getattr(col, operator)(value)

query = query.filter(clause)
```
where `model_name` is the name of the ORM class (at it assumes that all ORM classes are imported in the module that has this code)
